### PR TITLE
Set invoice labels after fedimint payment

### LINKED ISSUE #963: Making a payment to a contact from fedimint balance doesn't update contact's `last_updated`

### PR DESCRIPTION
**Pull request by benthecarman**
Closes #963

### CODE REVIEW

File: mutiny-core/src/lib.rs [hunk 1]
```diff
@@ -816,7 +816,10 @@ impl<S: MutinyStorage> MutinyWallet<S> {
                         .pay_invoice(inv.clone(), labels.clone())
                         .await;
                     match payment_result {
-                        Ok(r) => return Ok(r),
+                        Ok(r) => {
+                            self.storage.set_invoice_labels(inv.clone(), labels)?;
```
Comment (TonyGiorgio) [inline mutiny-core/src/lib.rs:820]: If it happens inside of `pay_invoice` for `node_manager` then it should happen inside `pay_invoice` for `fedimint_client`, or vice versa.

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]

File: mutiny-core/src/lib.rs (hunk 1 above)
Comment (benthecarman) [reply mutiny-core/src/lib.rs:820]: That's what I wanted but the fedimint client doesn't have an instance of storage.

Comment (TonyGiorgio) [reply mutiny-core/src/lib.rs:820]: Okay then they can both happen outside for now, that's probably better anyways so they both behave the same.

Comment (benthecarman) [reply mutiny-core/src/lib.rs:827]: Good call, did that for invoice creation as well. Cleaned up code a lot

**Pull request merged**